csharp/Adapter/Microsoft.Spark.CSharp/Interop/Ipc/JvmObjectReference.cs

```diff
@@ -21,6 +21,7 @@ public JvmObjectReference(string jvmReferenceId)
         {
             Id = jvmReferenceId;
             creationTime = DateTime.UtcNow;
+            WeakObjectManager.GetWeakObjectManager().AddWeakRefereceObject(this);
```

Comment:
GetWeakObjectManager() does not have any other reference, So, would it be better to delete that method and make AddWeakRefereceObject() a internal static method that internally uses the static member WeakObjManager directly?